yoticlient.go

```diff
@@ -104,65 +102,14 @@ func getActivityDetails(requester httpRequester, encryptedToken, sdkID string, k
 	}
 
 	if response.Success {
-		var parsedResponse = profileDO{}
-
-		if err = json.Unmarshal([]byte(response.Content), &parsedResponse); err != nil {
-			errStrings = append(errStrings, err.Error())
-			return
-		}
-
-		if parsedResponse.Receipt.SharingOutcome != "SUCCESS" {
-			err = ErrSharingFailure
-			errStrings = append(errStrings, err.Error())
-		} else {
-			var attributeList *yotiprotoattr.AttributeList
-			if attributeList, err = decryptCurrentUserReceipt(&parsedResponse.Receipt, key); err != nil {
-				errStrings = append(errStrings, err.Error())
-				return
-			}
-			id := parsedResponse.Receipt.RememberMeID
-
-			userProfile = addAttributesToUserProfile(id, attributeList) //deprecated: will be removed in v3.0.0
-
-			profile := Profile{
-				attributeSlice: createAttributeSlice(attributeList),
-			}
-
-			var formattedAddress string
-			formattedAddress, err = ensureAddressProfile(profile)
-			if err != nil {
-				log.Printf("Unable to get 'Formatted Address' from 'Structured Postal Address'. Error: %q", err)
-			} else if formattedAddress != "" {
-				if _, err = profile.StructuredPostalAddress(); err != nil {
-					errStrings = append(errStrings, err.Error())
-					return
-				}
-
-				protoStructuredPostalAddress := getProtobufAttribute(profile, attrConstStructuredPostalAddress)
-
-				addressAttribute := &yotiprotoattr.Attribute{
-					Name:        attrConstAddress,
-					Value:       []byte(formattedAddress),
-					ContentType: yotiprotoattr.ContentType_STRING,
-					Anchors:     protoStructuredPostalAddress.Anchors,
-				}
-
-				profile.attributeSlice = append(profile.attributeSlice, addressAttribute)
-			}
+		return handleSuccessfulResponse(response.Content, key)
```

Comment:
Might it be clearer if you assign `userProfile` and `activityDetails` here, as this method always returns `userProfile, activityDetails, errStrings`?

Reply:
As discussed, would be clearer to assign ActivityDetails here, instead of within `handleSuccessfulResponse`, but not worth doing whilst we have the old profile to deal with within that method. Will look at doing that once we do the major release.